src/codeu/chat/server/JSON.java

```diff
@@ -0,0 +1,110 @@
+package codeu.chat.server;
+
+import java.io.File;
+import java.io.IOException;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+
+import com.fasterxml.jackson.core.JsonGenerationException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public final class JSON {
+
+	private static void createNewJSON(ObjectMapper mapper) {
+		ObjectNode outerNode = mapper.createObjectNode();
+		outerNode.set("users", mapper.createArrayNode());
+		outerNode.set("conversations", mapper.createArrayNode());
+		outerNode.set("messages", mapper.createArrayNode());
+		ObjectWriter writer = mapper.writer(new DefaultPrettyPrinter());
+		try {
+			writer.writeValue(new File("data.json"), outerNode);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * Appends a user to the json file
+	 * 
+	 * @param user
+	 */
+	public static void save(User user) {
```

Comment:
It might be okay that these functions are static, but I wanted to double check that they're static for the right reasons. Basically: static should not be used to make it easier to access stuff, but it should be used if you'll never want to instantiate a class to call a function.

It's hard to know that you'll *never* do something. What if, for example, later I wanted to store the log in two separate files? (Maybe the primary file like you're using now and also to a backup file somewhere else.) Using static functions makes that harder. Instead, you'd probably want to use non-static functions in an instance that takes the file location as a constructor parameter.

There isn't a single right or wrong answer, so go with what you feel is best.